frontend/src/features/properties/map/activity/detail/ActivityPropertyModal.tsx

```diff
@@ -32,59 +32,76 @@ export const ActivityPropertyModal: React.FunctionComponent<IActivityPropertyMod
+
+  const handleCancel = () => {
+    if (
+      !dequal(
+        originalSelectedProperties.map(op => op.id),
+        selectedFileProperties.map(fp => fp.id),
+      )
+    ) {
+      setModalContent({
+        ...getCancelModalProps(),
+        handleOk: () => {
+          setSelectedFileProperties(originalSelectedProperties);
+          setDisplay(false);
+          setDisplayModal(false);
+        },
+      });
+      setDisplayModal(true);
+    } else {
+      setDisplay(false);
+      setDisplayModal(false);
+    }
+  };
+
+  const handleOk = async () => {
+    if (activityModel && allProperties?.length > 0) {
+      activityModel.actInstPropFiles =
+        selectedFileProperties.map(p => ({
+          activityId: activityModel?.id,
+          propertyFileId: p?.id,
+        })) ?? [];
+      const updatedActivity = await onSave(activityModel);
+      // if the activity was updated successfully, hide the modal.
+      if (!!updatedActivity) {
+        setDisplay(false);
+      }
+    }
+  };
+
   return (
     <GenericModal
       modalSize={ModalSize.MEDIUM}
       title="Related properties"
       display={display}
       setDisplay={setDisplay}
       closeButton={true}
-      okButtonText="Save"
-      cancelButtonText="Cancel"
-      handleCancel={() => {
-        if (
-          !dequal(
-            originalSelectedProperties.map(op => op.id),
-            selectedFileProperties.map(fp => fp.id),
-          )
-        ) {
-          setModalContent({
-            ...getCancelModalProps(),
-            handleOk: () => {
-              setSelectedFileProperties(originalSelectedProperties);
-              setDisplay(false);
-              setDisplayModal(false);
-            },
-          });
-          setDisplayModal(true);
-        } else {
-          setDisplay(false);
-          setDisplayModal(false);
-        }
-      }}
-      handleOk={async () => {
-        if (activityModel) {
-          activityModel.actInstPropFiles =
-            selectedFileProperties.map(p => ({
-              activityId: activityModel?.id,
-              propertyFileId: p?.id,
-            })) ?? [];
-          const updatedActivity = await onSave(activityModel);
-          // if the activity was updated successfully, hide the modal.
-          if (!!updatedActivity) {
-            setDisplay(false);
-          }
-        }
-      }}
-      message={
-        <PropertyActivityTable
-          fileProperties={allProperties ?? []}
-          selectedFileProperties={selectedFileProperties}
-          setSelectedFileProperties={setSelectedFileProperties}
-        />
-      }
+      okButtonText={allProperties?.length > 0 ? 'Save' : 'Ok'}
+      cancelButtonText={allProperties?.length > 0 ? 'Cancel' : undefined}
+      handleCancel={allProperties?.length > 0 ? handleCancel : undefined}
+      handleOk={allProperties?.length > 0 ? handleOk : undefined}
+      message={renderModalContent(allProperties, selectedFileProperties, setSelectedFileProperties)}
```

Comment:
nit the function does not actually render, but rather creates the sub view. In that light, what do you think about making that a proper React component instead?

Reply:
sure, works for me